src/js/models/characterstyle.js

```diff
@@ -187,64 +140,67 @@ define(function (require, exports, module) {
-            }
-        } else {
-            if (previousResult.leadingValid === null || previousResult.leading === null) {
-                model.leadingValid = true;
-            } else {
-                model.leading = undefined;
-                model.leadingValid = false;
-            }
+
+            model.leading = leading;
         }
 
         return new CharacterStyle(model);
     };
 
     /**
-     * Construct a list of new CharacterStyle models from the given descriptor, or return null
+     * Construct a reduce CharacterStyle model from the given descriptor, or return null
      * if there are no character styles, e.g., if the descriptors do not describe a text layer.
+     * The CharacterStyle model will have undefined properties in case the models of the
+     * individual textStyleRange descriptors are inconsistent.
      * 
      * @static
      * @param {object} documentDescriptor
      * @param {object} layerDescriptor
      * @param {object} textDescriptor
-     * @return {?Immutable.List.<CharacterStyle>}
+     * @return {CharacterStyle}
      */
     CharacterStyle.fromTextDescriptor = function (documentDescriptor, layerDescriptor, textDescriptor) {
         var textStyleRanges = textDescriptor.textStyleRange;
 
-        // Only the first style range contains the baseParentStyle
-        var firstBaseParentStyle;
-        if (textStyleRanges.length > 0) {
-            firstBaseParentStyle = objUtil.getPath(textStyleRanges[0],
-                "textStyle.baseParentStyle");
+        if (!textStyleRanges || textStyleRanges.length === 0) {
+            return new CharacterStyle();
         }
 
-        return Immutable.List(textStyleRanges)
-            .reduce(function (result, characterStyleDescriptor) {
+        // Only the first style range contains the baseParentStyle
+        var firstTextStyleRange = textStyleRanges[0],
+            firstBaseParentStyle = objUtil.getPath(firstTextStyleRange, "textStyle.baseParentStyle");
+
+        var characterStyles = Immutable.List(textStyleRanges)
+            .map(function (characterStyleDescriptor) {
                 var baseParentStyle = objUtil.getPath(characterStyleDescriptor,
                         "textStyle.baseParentStyle");
 
                 baseParentStyle = baseParentStyle || firstBaseParentStyle;
 
-                return CharacterStyle.fromCharacterStyleDescriptor(documentDescriptor, layerDescriptor,
-                    characterStyleDescriptor, baseParentStyle, result);
-            }, new CharacterStyle());
+                return CharacterStyle.fromCharacterStyleDescriptor(documentDescriptor,
+                    layerDescriptor, characterStyleDescriptor, baseParentStyle);
+            });
+
+        var reducedModel = Object.keys(_characterStylePrototype)
+            .reduce(function (reducedModel, property) {
+                var values = collection.pluck(characterStyles, property);
+
+                reducedModel[property] = collection.uniformValue(values);
```

Comment:
In the case of these reduce models, null will be interpreted in the UI to mean "mixed".